index/vector_index.cc

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2025-present ScyllaDB
+ *
+ * Modified by ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: (LicenseRef-ScyllaDB-Source-Available-1.0 and Apache-2.0)
+ */
+
+
+#include "cql3/statements/index_target.hh"
+#include "exceptions/exceptions.hh"
+#include "schema/schema.hh"
+#include "index/vector_index.hh"
+#include "concrete_types.hh"
+#include <seastar/core/sstring.hh>
+
+
+namespace secondary_index {
+
+template <int MAX>
+void validate_unsigned_option(const sstring& value) {
+    int num_value;
+    size_t len;
+    try {
+        num_value = std::stoi(value, &len);
+    } catch (...) {
+        throw exceptions::invalid_request_exception(format("Numeric option {} is not a valid number", value));
+    }
+    if (len != value.size()) {
+        throw exceptions::invalid_request_exception(format("Numeric option {} is not a valid number", value));
+    }
+
+    if (num_value < 0 || num_value > MAX) {
+        throw exceptions::invalid_request_exception(format("Numeric option {} out of valid range [0 - {}]", value, MAX));
```

Comment:
Oh, I just realized that you actually allow MAX itself, so the error message is correct. Just please verify that you intentionally want to allow MAX and not just MAX-1 (I don't really care, both are fine, I just don't know what your intention was).